contrib/insights/src/column_statistics.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You
+ * may not use this file except in compliance with the License. A copy of
+ * the License is located at
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "mlio.h"
+#include "hll/hyperloglog.h"
+
+namespace insights {
+namespace detail {
+
+static const uint8_t CARDINALITY_HLL_SIZE = 16;
+class MLIO_API column_analysis {
+    public:
+
+    column_analysis(std::string name) 
+        : rows_seen(0),
+          numeric_mean(0.0), 
+          numeric_count(0), 
+          numeric_nan_count(0), 
+          numeric_min(std::numeric_limits<double>::quiet_NaN()),
+          numeric_max(std::numeric_limits<double>::quiet_NaN()),
+          string_min_length(std::numeric_limits<size_t>::max()),
+          string_min_length_not_empty(std::numeric_limits<size_t>::max()),
+          string_max_length(0),
+          string_empty_count(0),
+          string_only_whitespace_count(0),
+          string_null_like_count(0),
+          string_captured_unique_values_overflowed(false),
+          example_value("")
+    { column_name = name; 
+      string_cardinality_estimator = hll::HyperLogLog(CARDINALITY_HLL_SIZE); 
+    }
+
+    size_t string_cardinality() const {
+      return static_cast<size_t>(std::round(string_cardinality_estimator.estimate()));
```

Comment:
Is this an expensive operation? Should we cache it? Right now the signature of this function resembles an accessor; if the cost of the `estimate()` call is high, we should either cache it beforehand; or change the signature of this function to something like `estimate_string_cardinality()`.